crates/ott-balancer/src/selection.rs

```diff
@@ -49,34 +49,72 @@ mod test {
     #[tokio::test]
     async fn test_min_by() {
-        let _room_name_one = RoomName::from("room one");
-        let _room_name_two = RoomName::from("room two");
-        let _room_name_three = RoomName::from("room three");
+        let mut ctx = BalancerContext::new();
+        let room_one = RoomName::from("room one");
+        let room_two = RoomName::from("room two");
+        let room_three = RoomName::from("room three");
         let (monolith_outbound_tx, _monolith_outbound_rx) = tokio::sync::mpsc::channel(100);
-        let monolith_outbound_tx = Arc::new(monolith_outbound_tx);
-        let (client_inbound_tx, _client_inbound_rx) = tokio::sync::mpsc::channel(100);
-        let monolith_id = uuid::Uuid::new_v4().into();
+        let monolith_outbound_tx_one = Arc::new(monolith_outbound_tx);
+        let (client_inbound_tx_one, _client_inbound_rx) = tokio::sync::mpsc::channel(100);
+        let monolith_id_one = uuid::Uuid::new_v4().into();
 
-        let _monolith_one = BalancerMonolith::new(
+        let monolith_one = BalancerMonolith::new(
             NewMonolith {
-                id: monolith_id,
+                id: monolith_id_one,
                 region: "unknown".into(),
                 config: MonolithConnectionConfig {
                     host: HostOrIp::Ip(Ipv4Addr::LOCALHOST.into()),
                     port: 3002,
                 },
                 proxy_port: 3000,
             },
-            monolith_outbound_tx,
-            client_inbound_tx,
+            monolith_outbound_tx_one,
+            client_inbound_tx_one,
         );
+
+        let (monolith_outbound_tx, _monolith_outbound_rx) = tokio::sync::mpsc::channel(100);
+        let monolith_outbound_tx_two = Arc::new(monolith_outbound_tx);
+        let (client_inbound_tx_two, _client_inbound_rx) = tokio::sync::mpsc::channel(100);
+        let monolith_id_two = uuid::Uuid::new_v4().into();
+
+        let monolith_two = BalancerMonolith::new(
+            NewMonolith {
+                id: monolith_id_two,
+                region: "unknown".into(),
+                config: MonolithConnectionConfig {
+                    host: HostOrIp::Ip(Ipv4Addr::LOCALHOST.into()),
+                    port: 3002,
+                },
+                proxy_port: 3000,
+            },
+            monolith_outbound_tx_two,
+            client_inbound_tx_two,
+        );
+
+        ctx.add_monolith(monolith_one);
+        ctx.add_room(room_one.clone(), RoomLocator::new(monolith_id_one, 0))
+            .expect("failed to add room");
+        ctx.add_room(room_two.clone(), RoomLocator::new(monolith_id_one, 0))
+            .expect("failed to add room");
+
+        ctx.add_monolith(monolith_two);
+        ctx.add_room(room_three.clone(), RoomLocator::new(monolith_id_two, 0))
+            .expect("failed to add room");
+
+        let filtered = ctx.filter_monoliths();
```

Comment:
You shouldn't be calling this in this test. The goal here is to test `MinRoomsSelector`. Any impl of `MonolithSelection`  only operates on a set of `BalancerMonolith`s.